service/server.go

```diff
@@ -7,17 +7,18 @@ import (
 
 	metrics "github.com/grpc-ecosystem/go-grpc-middleware/providers/openmetrics/v2"
 	grpczerolog "github.com/grpc-ecosystem/go-grpc-middleware/providers/zerolog/v2"
+	"github.com/grpc-ecosystem/go-grpc-middleware/v2/interceptors/auth"
 	"github.com/grpc-ecosystem/go-grpc-middleware/v2/interceptors/logging"
 	"github.com/grpc-ecosystem/go-grpc-middleware/v2/interceptors/recovery"
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/rs/zerolog"
 	"google.golang.org/grpc"
 )
 
-// NewGrpcService creates a grpc service with various defaults middlewares.
+// NewGRPCService creates a grpc service with various defaults middlewares.
 // Notably, the logging and metrics are automatically registered for sane
 // defaults of observability.
-func NewGrpcService(ctx context.Context, service interface{}, descriptors []*grpc.ServiceDesc) (*grpc.Server, error) {
+func NewGRPCService(ctx context.Context, service interface{}, authFunc auth.AuthFunc, descriptors []*grpc.ServiceDesc) (*grpc.Server, error) {
```

Comment:
I believe it would be better to provide an *optional* list of interceptors. If none is provided, default to the list provide down here.